oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java

```diff
@@ -451,7 +457,23 @@ private HttpResponse getMetadataResponse(
     request.setThrowExceptionOnExecuteError(false);
     HttpResponse response;
     try {
+      String requestMessage;
+      String responseMessage;
+      if (requestType.equals(RequestType.ID_TOKEN_REQUEST)) {
+        requestMessage = "Sending request to get ID token";
+        responseMessage = "Received response for ID token request";
+      } else if (requestType.equals(RequestType.ACCESS_TOKEN_REQUEST)) {
+        requestMessage = "Sending request to refresh access token";
+        responseMessage = "Received response for refresh access token";
+      } else {
+        // TODO: this includes get universe domain and get default sa.
+        // refactor for more clear logging message.
+        requestMessage = "Sending request for universe domain/default service account";
+        responseMessage = "Received response for universe domain/default service account";
```

Comment:
qq, I see that this takes in a `shouldSendMetricsHeader` header.

Thoughts on perhaps creating our internal RequestType values (one for Universe Domain and one for Service Account). It could differentiate between UD and SA requests, but also not track the metrics header that can be used for logging.

Reply:
yea, I think that's a valid improvement. However, given the size of this PR, I'd rather file as a separate issue as clean-up/enhancement.